src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -357,7 +356,12 @@ void WlSurfaceCursor::apply_latest_buffer()
             auto const cursor_image = std::make_shared<BufferCursorImage>(
                 *stream->lock_compositor_buffer(this),
                 hotspot);
-            surface->set_cursor_image(cursor_image);
+            if (cursor_image->as_argb_8888())
+                surface->set_cursor_image(cursor_image);
+            else
+            {
+
+            }
```

Comment:
This looks awkward. Can you either drop the else block or put something in it (and make braces/lack thereof consistent)?